lib/graphql/query/variable_validation_error.rb

```diff
@@ -25,6 +30,15 @@ def to_h
           end
         end
       end
+
+      private
+
+      def problem_fields
+        @problem_fields ||= @validation_result
+          .problems
+          .reject { |problem| problem["path"].empty? }
+          .map { |problem| problem["path"].join(".") }
```

Comment:
I think it'd be super useful to include the `explanation` here. So for the modified test cases below, you'd end up with messages of the form:

```"message" => "Variable input of type ReplaceValuesInput! was provided invalid value for values (Expected value to not be null)"```
and
```"message" => "Variable input of type [DairyProductInput] was provided invalid value for 0.foo (Field is not defined on DairyProductInput), 0.source (Expected value to not be null)",```

I know this is duping information from the problems hash, but I feel just the field name alone, especially in an array case of multiple issues, not knowing why it was invalid could still hinder debugging.

What do you both think to this?

Reply:
My only reservation is the messages can be quite long now but I'm into it. Updated in 8a8e189be.